src/binder/bind_expression/bind_comparison_expression.cpp

```diff
@@ -26,35 +25,31 @@ std::shared_ptr<Expression> ExpressionBinder::bindComparisonExpression(
     ExpressionType expressionType, const expression_vector& children) {
     auto functions = context->getCatalog()->getFunctions(binder->clientContext->getTx());
     auto functionName = expressionTypeToString(expressionType);
-    std::vector<LogicalType> childrenTypes;
+    LogicalType combinedType(LogicalTypeID::ANY);
+    bool canCombine = true;
     for (auto& child : children) {
-        childrenTypes.push_back(child->dataType);
+        if (!LogicalTypeUtils::tryGetMaxLogicalType(combinedType, child->dataType, combinedType)) {
+            canCombine = false;
+        }
+    }
+    if (!canCombine) {
```

Comment:
Is this the expect behaviour? I thought it would be dangerous if we always succeed in binding comparison. Throw exception seems to be more reasonable here.